clippy_lints/src/matches.rs

```diff
@@ -543,14 +554,21 @@ impl_lint_pass!(Matches => [
     MATCH_SAME_ARMS,
 ]);
 
+const MATCH_LIKE_MATCHES_MACRO_MSRV: &str = "1.42.0";
+
 impl<'tcx> LateLintPass<'tcx> for Matches {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
         if in_external_macro(cx.sess(), expr.span) || in_macro(expr.span) {
             return;
         }
 
         redundant_pattern_match::check(cx, expr);
-        if !check_match_like_matches(cx, expr) {
+
+        if self.msrv.is_none() || meets_msrv(self.msrv.as_ref(), MATCH_LIKE_MATCHES_MACRO_MSRV) {
```

Comment:
You should be able to remove the `self.msrv.is_none()` part here, since `meets_msrv` should return `true` if `msrv` is `None`.

Reply:
Ah right, I'll update this